src/daqlite/ThreadSafeVector.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2024 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file CustomAMOR2DTOFPlot.h
+///
+/// \brief Creates (maybe) a QCustomPlot based on the configuration parameters
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cstdint>
+#include <map>
+#include <mutex>
+#include <vector>
+
+template <typename T, typename R> class ThreadSafeVector {
+public:
+  void push_back(const T &value) {
+    std::lock_guard<std::mutex> lock(mMutex);
+    mVector.push_back(value);
+  }
```

Comment:
At least the template should have some doxygen comments describing what it does and what T and especially R are.

Reply:
documentation added